contracts/src/AccountRegistry.sol

```diff
@@ -247,23 +353,21 @@ contract AccountRegistry is Initializable, EIP712Upgradeable, Ownable2StepUpgrad
         uint256[] calldata authenticatorPubkeys,
         uint256 offchainSignerCommitment
     ) internal virtual {
-        require(authenticatorAddresses.length > 0, "authenticatorAddresses length must be greater than 0");
-        require(
-            authenticatorAddresses.length == authenticatorPubkeys.length,
-            "authenticatorAddresses and authenticatorPubkeys length mismatch"
-        );
-        require(recoveryAddress != address(0), "Recovery address cannot be the zero address");
```

Comment:
@philsippl FYI, removing this. the recovery address can be nil for users who opt out